compiler/ilgen/IlBuilder.cpp

```diff
@@ -1494,9 +1494,8 @@ IlBuilder::integerizeAddresses(TR::IlValue **leftPtr, TR::IlValue **rightPtr)
    {
    TR::IlValue *left = *leftPtr;
    TR::IlValue *right = *rightPtr;
-   if (left->getSymbol()->getDataType() == TR::Address)
+   if (left->getSymbol()->getDataType() == TR::Address && right->getSymbol()->getDataType() == TR::Address)
```

Comment:
From a usability point of view, I'm not sure I like the idea of this silently failing if the two IlValues are not addresses. I'm thinking of a scenario where I accidentally pass something that's an address and another thing that isn't. The call will silently fail to convert the address, causing some assert to fail later on because an operation is not compatible with addresses. Then, I get really confused because I've clearly called `integerizeAddresses` but I'm still getting an address.